src/main/java/build/buildfarm/worker/Worker.java

```diff
@@ -222,17 +225,21 @@ private boolean fetchInputs(Path execDir, Digest inputRoot) throws IOException {
     ImmutableList.Builder<Directory> directories = new ImmutableList.Builder<>();
     String pageToken = "";
 
-    Map<Digest, Directory> directoriesIndex = new HashMap<Digest, Directory>();
-
     do {
       pageToken = instance.getTree(inputRoot, config.getTreePageSize(), pageToken, directories);
     } while (!pageToken.isEmpty());
 
-    directoriesIndex = Maps.uniqueIndex(
-        directories.build(),
-        directory -> Digests.computeDigest(directory));
+    Set<Digest> directoryDigests = new HashSet<>();
+    ImmutableMap.Builder<Digest, Directory> directoriesIndex = new ImmutableMap.Builder<>();
```

Comment:
As in two entries in the DirectoryNode list (or FileNode list for that matter) with the same name? AbstractServerInstance#validateActions from #1 (and #validateActionInputs refactored in #14) has a precondition check for duplicate file/directory nodes (considers them in one Set). I will be refactoring those validations out into a common Directories (a la Digests) toolset for validating them both on the instance server and the worker in upcoming changes.